File: Irrgarten_Ruby/Monster.rb
#encoding:utf-8

require_relative 'Dice'

module Irrgarten 
	class Monster 
		@@INITIAL_HEALTH = 5
		
		
		def initialize ( name, intelligence, strength) #string,float,float
			@name = name
			@intelligence = intelligence
			@strength = strength
			@health = @@INITIAL_HEALTH
			@col = -1
			@row = -1
		end
		
		def dead () 
			if @health <= 0
				return true
			else
				return false
			end
		
		end
		
		def attack()
			return Dice.intensity(@strength)
		end
		
		def defend( received_attack ) #float
			#se completa mas tarde
			
		end
		
		def set_pos ( row, col) # int, int
			if row >= 0 && col >=0
				@row=row
				@col=col
			else 
				puts "El monstruo no puede estar en una posición negativa"
			end
			
		end
		
		def got_wounded()
			@health = @health - 1
		
		end
		
		def to_s()
			"Name: #{@name}, Intelligence: #{@intelligence}, Strength: #{@strength}, Health: #{@health}, Position: (#{@row}, #{@col})"
		end
			
		
	end
end

File: Irrgarten_Java/Irrgarten_P3/src/irrgarten/Monster.java

package irrgarten;

/**
 *
 * @author leonin04
 */
public class Monster {
    private static final int INITIAL_HEALTH = 5;
    
    private String name;
    private float intelligence;
    private float strength;
    private float health;
    private int row;
    private int col;
    
    public Monster (String name, float intelligence, float strength){
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        health = INITIAL_HEALTH;
        row = -1;
        col = -1; //Con esto representamos que no está colocado
    }
    
    public boolean dead(){
        if (health <= 0){
            return true;
        } else return false;
    }
    
    public float attack(){
        return Dice.intensity(strength);
    }
    
    public boolean defend(float receivedAttack){
        boolean isDead = dead();
        if(isDead == false){
            float defensiveEnergy = Dice.intensity(intelligence);
            if (defensiveEnergy < receivedAttack){
                gotWounded();
                isDead = this.dead();
            }
            
        }
        return isDead;
    }  
    
    public void setPos(int row, int col){
        assert row >= 0 && col >= 0 : "El monstruo no puede estar en una posición negativa";
        
        this.row = row;
        this.col = col;
    }
    
    public void gotWounded(){
        health--;
    }
    
    @Override
    public String toString(){
        return ("Name: " + name + ", Intelligence: " + intelligence + ", Strength: " + strength + ", Health: " + health + ", Position: (" + row + "," + col + ")\n");
    }
}
